### DISCUSSION TITLE
Condividere apertura del cancello

### DISCUSSION THREAD

**Discussion by orgnemo** [indomus/forum#1248] [HUB: Home Assistant]
Salve a tutti,
vorrei condividere l'apertura del cancello con delle persone di famiglia. Usando gli utenti di HA ho notato che non si possono disabilitare molte funzionalità come la voce mappe, la voce energia etc.
Voi come gestite queste situazioni? Avevo pensato ad una url che con credenziali che richiamandola apre il cancello.

Grazie

**Comment by giacomo458** [comment]
Ciao io ho condiviso con i vicini tramite blynk e nodered

**Reply by orgnemo** [reply]
grazie per la risposta

**Reply by jimborm** [reply]
Ciao io ho creato un bot Telegram in cui ho condiviso ai condomini l'apertura del cancello del residence e agli amici ad esempio anche quella del box quando vengono a trovarmi.

**Reply by orgnemo** [reply]
grazie per la risposta, ho fatto un test con webhook di home assistant e un app android "http shortcats" sembra funzionare bene, ho pensato di farne una per ogni persona a cui intendo condividerla in modo da tracciare abusi e utilizzare GUID complessi.

**Reply by jimborm** [reply]
Per mia curiosità mi diresti più nel dettaglio la tua soluzione?

**Reply by orgnemo** [reply]
Si certamente,
in questi giorni stavo configurando un reverse proxy e ho notato nei log qualcosa tipo questo:
```
5.168.40.21 - - [07/Jan/2022:19:56:10 +0100] "POST /api/webhook/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX HTTP/1.1" 200 2 "-" "HomeAssistant/Android SM-xxxx 2022.1.1-1890"

```
quindi sono andato a vedere e ho immaginato che l'app di Home assistan companion, in alcuni casi, comunica con gli web hook. La configurazione è semplice perchè basta fare una automazione tipo questa:
```
alias: Apri il cancello con webhook
description: ''
trigger:
  - platform: webhook
    webhook_id: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
condition: []
action:
  - service: switch.turn_on
    target:
      entity_id: switch.cancello_di_casa
mode: single

```

Chiunque richiami la url https://remote_ip:8123/api/webhook/xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx in POST può aprire il cancello, ecco perchè è importante avere una [GUID](https://www.guidgenerator.com/online-guid-generator.aspx) molto complessa al posto delle X.

A questo punto ho trovato questa app "http shortcuts" che ha molte funzioni con le url, tra cui fare una icona come se fosse una app da aggiungere alla schermata iniziale. 
Questi web hook aprono scenari infiniti e soprattutto puoi scegliere quali funzionalità condividere. Io avevo pensato di fare una automazione per persona a cui voglio condividere, in modo che se ne vedo un abuso disattivo solo quella persona.
Il bot telegram lo uso come log di automazioni, per il citofono, mi manda una istantanea della telecamera quando suonano, e cose del genere. 
Per le persone un pò in la' con l'età sarebbe stato complicato usare telegram, in oltre ne avrei dovuto usarne uno diverso.
Spero di essere stato chiaro.

**Reply by jimborm** [reply]
Si capito a grandi linee crei una URL, contenente una GUID molto complessa, che corrisponde alla tua webhook e che fa scatenare l'automazione. Poi con "http shortcuts" crei sul desktop del telefono (Android e iOS?) un icona che quando tappata carica la url sul browser scatenando l'automazione. 

Anche io il bot lo uso come "log" dei tentativi di apertura del cancello solo volevo anche contare quante volte viene aperto ma sono talmente scarso che sto ancora cercando di capire come mettere un conta aperture :-( ho anche aperto sul forum una discussione ma nessuno mi ha ancora risposto.
 Il reverse PROXY ho provato ad installarlo ma naturalmente non funzionava :-)
